.github/workflows/test-python.yml

```
@@ -62,6 +63,39 @@ jobs:
     - name: Run simple code
       run: python -c 'import math; print(math.factorial(5))'
 
+  setup-versions-from-file:
+    name: Setup ${{ matrix.python }} ${{ matrix.os }} version file
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-18.04, ubuntu-20.04]
+        python: [3.5.4, 3.6.7, 3.7.5, 3.8.1]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: build-version-file ${{ matrix.python }}
+      run: echo ${{ matrix.python }} > .python-version
+
+    - name: setup-python ${{ matrix.python }}
+      uses: ./
+      with:
+        python-version-file: '.python-version'
```

I would expect to specify this external version source like so
```suggestion
    - name: setup-python (version from file)
      uses: ./
      with:
        python-version: 'file'
```


With file name being customizable and default=`.python-version`
```suggestion
    - name: setup-python ${{ matrix.python }}
      uses: ./
      with:
        python-version: 'file'
        python-version-file: '.custom-python-version'
```

---

Resolved